packages/provider-elasticsearch/src/example-types/results/highlighting/response.js

```diff
@@ -0,0 +1,189 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import { isArraysEqual, groupByIndexed } from '../../../utils/futil.js'
+import {
+  stripTags,
+  mergeHighlights,
+  isBlobField,
+  isArrayField,
+  isArrayOfScalarsField,
+  findByPrefix,
+  getArrayOfObjectsPathsMap,
+} from './util.js'
+
+let lastWordRegex = /\.(\w+)$/
+
+/*
+ * Group highlight results by their multifield. For example `city` and
+ * `city.subfield` will be grouped under `city`.
+ */
+let groupByMultiField = _.curry((schema, highlight) =>
+  groupByIndexed((v, path) => {
+    let [multi, sub] = path.split(lastWordRegex)
+    return schema.fields[multi]?.elasticsearch?.mapping?.fields?.[sub]
+      ? multi
+      : path
+  }, highlight)
+)
+
+/**
+ * Mutate hit `highlight`:
+ * 1. Fragments for text fields get merged.
+ * 2. Fragments for large (blob) text fields get concatenated.
+ * 3. Fragments for arrays get ordered based on the source array
+ */
+export let transformResponseHighlight = (
+  schema,
+  hit,
+  tags,
+  nestedArrayIncludes = {}
+) => {
+  let arrayOfObjectsPaths = _.keys(getArrayOfObjectsPathsMap(schema))
+
+  let getIndexedArrayObject = (arr, fragments, arrayPath, itemPath) => {
+    let fragmentsMap = _.groupBy(stripTags(tags), fragments)
+    return F.reduceIndexed(
+      (acc, item, index) => {
+        let fragments = fragmentsMap[itemPath ? _.get(itemPath, item) : item]
+        if (fragments) {
+          F.setOn(
+            F.dotJoin([`${index}`, itemPath]),
```

Comment:
add a comment about why we're casting to a string (to avoid compacting 0 indexes, but still wanting to omit an empty itemPath)

Reply:
Even with a comment this is a good example of code that is hard to follow because it requires you to know that `F.dotJoin` also does compaction. Something that is not obvious from the name `dotJoin`.

Reply:
This code doesn't exist anymore.